TLM/TLM/UI/DebugSwitches/DebugSwitchPanel.cs

```diff
@@ -0,0 +1,134 @@
+#if DEBUG
+namespace TrafficManager.UI.DebugSwitches {
+    using ColossalFramework.UI;
+    using CSUtil.Commons;
+    using System;
+    using System.Collections.Generic;
+    using TrafficManager.State.ConfigData;
+    using TrafficManager.U;
+    using UnityEngine;
+    using TrafficManager.Util;
+
+    public class DebugSwitchPanel : UIPanel {
+        private const int _defaultWidth = 400;
+        private const int _defaultHeight = 700;
+        private static readonly Color32 _panelBgColor = new Color32(55, 55, 55, 255);
+        private static readonly RectOffset _panelPadding = new RectOffset(10, 0, 15, 0);
+
+        private static List<DebugSwitchCheckboxOption> options_ = new();
+
+        private UIDragHandle _header;
+
+        static DebugSwitchPanel() {
+            foreach (DebugSwitch debugSwitch in Enum.GetValues(typeof(DebugSwitch))) {
+                if (debugSwitch != DebugSwitch.None) {
+                    options_.Add(new DebugSwitchCheckboxOption(debugSwitch));
+                }
+            }
+        }
+
+        public override void Awake() {
+            base.Awake();
+
+            isVisible = true;
+            canFocus = true;
+            isInteractive = true;
+            anchor = UIAnchorStyle.Left | UIAnchorStyle.Top | UIAnchorStyle.Proportional;
+            size = new Vector2(_defaultWidth, _defaultHeight);
+            backgroundSprite = "GenericPanel";
+            color = _panelBgColor;
+            atlas = TextureUtil.Ingame;
+
+            AddHeader();
+            AddContent();
+        }
+
+        public static void OpenModal() {
```

Comment:
Q: Is there a chance to trigger this action without closing the modal that was opened before?
If yes, then there should be a check in UIView if panel exist and most likely just abort or call bring to front.

Reply:
I don't see any chance. I copied this code from de-spawning panel you wrote. if there is a problem here there is a problem there too.

Reply:
ok, good, I also won't able to break it, so ignore :)